File: Striver/JAVA/implement_stack_with_queue.java
import java.util.LinkedList;
import java.util.Queue;

public class implement_stack_with_queue {
    public static class QueueClass{
        public Queue <Integer> qt = new LinkedList<>();
        public void push(int x){
            qt.add(x);
            for(int i=0;i<qt.size()-1;i++){
                qt.add(qt.remove());
            }
        }
        public int pop(){
            if(qt.isEmpty()){
                throw new RuntimeException("Stack Is Empty");
            }
            return qt.remove();
        }
        public int top(){
            if(qt.isEmpty()){
                throw new RuntimeException("Stack Is Empty");
            }
            return qt.peek();
        }
        public boolean isQueueEmpty(){
            return qt.isEmpty();
        }
    }
    public static void main(String[] args) {
        QueueClass stack = new QueueClass();
        stack.push(4);
        stack.push(9);
        stack.push(5);
        stack.push(2);
        System.out.println("The top of the stack is " + stack.top());
        stack.pop();
        stack.pop();
        System.out.println("The top of the stack is " + stack.top());
        stack.push(1);
        stack.top();
    }
}

File: Striver/CPP/implement_stack_with_queue.cpp
#include<iostream>
#include<queue>

using namespace std;

class QueueClass{
    queue<int> qt;
    public:
        void push(int x){
            qt.push(x);
            for(int i=0;i<qt.size()-1;i++){
                qt.push(qt.front());
                qt.pop();
            }
        }
        int pop(){
            if(qt.empty()){
                cout<<"Stack is empty"<<endl;
            }
            int ele = qt.front();
            qt.pop();
            return ele;
        }
        int top(){
            if(qt.empty()){
                cout<<"Stack is empty"<<endl;
            }
            return qt.front();
        }
        bool isEmpty(){
            return qt.empty();
        }
};

int main(){
    QueueClass stack;
    stack.push(4);
    stack.push(9);
    stack.push(5);
    stack.push(2);
    cout<<"Top of the stack is : "<<stack.top()<<endl;
    stack.pop();
    stack.pop();
    cout<<"Top of the stack is : "<<stack.top()<<endl;
    stack.push(1);
    cout<<"Top of the stack is : "<<stack.top()<<endl;
}